### DISCUSSION TITLE
Fl_TransBox to display information, and resizing tips.

### DISCUSSION THREAD

**Discussion by rageworx** [fire-eggs/YAIV#37] [Ideas]
![image](https://user-images.githubusercontent.com/6948225/120591975-fece0900-c477-11eb-96f3-b0f8fb8ef914.png)

* XBox is implement to Fl_Box, not Fl_Group.
* removed 5pixel borders ... ( modern GUI viewers mostly don't use border as I experienced )
* Using Fl_TransBox to displaying image information.
* Some resizing tips in changes, see below. ( using invisible Fl_Group and it used to be resizing method )

Check to my forked version, https://github.com/rageworx/YAIV

* You said before about glitches, it must be belong to you are drawing something in Fl_Wdiget::draw() implement....
* I believe it's similar as Fl_Window and Fl_Double_Window difference.
* It should be better if you are draw something in back buffer ( like Fl_RGB_Image ),
* then just draw back buffer in Fl_Widget::draw() time. 
* It will prevent flickering issue.
* In case of Windows GDI, it using double buffering by SwapBuffers( HDC ) it's same like uising back buffer with Fl_RGB_Image.

--- and ...
* I'm using SwapBuffers( HDC ) on OpenGL surface to preventing flickering screen.

**Comment by fire-eggs** [comment]
That looks pretty nice, I'll think about trying it out.

>It should be better if you are draw something in back buffer ( like Fl_RGB_Image ),

The current code does just that [setting aside animated images]. The "original" image is loaded in `_img`; the drawn image (back buffer) `_showImg`, is created by `XBox::update_image()`. That function does the following: 
- converts pixmaps (GIF) into an FL_RGB_Image
- rotation
- scaling and zoom
- draw the checkerboard background

So the `draw()` overload is "merely": 
- draw the `_showImg` 
- draw the minimap
- draw the overlay

The update_image function "should" only be called when the zoom, scale, checker is changed (or the original image). Drawing "should" be reasonably fast as the drawing image "should" not be rebuilt as the user pans about.

I use "should" because I'm seeing flicker as I scroll, even with the minimap and overlay OFF. Something doesn't seem to be as fast as I'd like.

**Reply by fire-eggs** [reply]
And of course the screen capture can't capture the flicker.

**Reply by fire-eggs** [reply]
[repeating myself]
So I've tried your code. Even with Fl_TransBox, minimap, and checkerboard turned off, I'm still seeing flicker. But I've also confirmed that updateImage() is called only once. [I.e. the back-buffer image is not rebuilt while the user pans the image].

This tells me the FLTK drawing (whether via FL_Group or FL_Box) is probably the problem.

**Reply by rageworx** [reply]
> This tells me the FLTK drawing (whether via FL_Group or FL_Box) is probably the problem.

I'll be look for back-buffer-draw ( or off-screen drawing to Fl_Image ) and draw it once for image->draw(x(),y()) as checking flickering.
